ui/scripts/modal.js

```diff
@@ -1,18 +1,29 @@
 export default class Modal {
     // @todo: will need to add trailer url later
-    constructor(title) {
+    constructor(title, trailer_url) {
         this.title = title;
-        // this.url = url;
+        this.trailer_url = trailer_url;
+    }
+
+    getId(url) {
+        const regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
+        const match = url.match(regExp);
+    
+        return (match && match[2].length === 11)
```

Comment:
what's "11"? please put it in a const with a self-explanatory name. Is it "id"? Is the id always guaranteed to be 11-characters? Can you please confirm it with youtube API docs? (Or where is the 11-char limit documented?)

Reply:
there's no info in there docs, the only thing I found was the reference to the nonexistent now post from a Youtube API Team that said:
> I don't see anywhere in the documentation where we officially commit to a standard length of 11 characters for YouTube video ids. It's one of those things where we have a current implementation, and it may stay that way indefinitely. But we're not offering any official commitment to that, so proceed at your own risk.

So it's probably a good idea just to extract whatever is there and do not expect an id to be 11-char long.